integration/suites/harness/agent_test.go

```diff
@@ -0,0 +1,152 @@
+)
+
+func TestAgentHarnessAgent(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "agent testing harness")
+}
+
+var _ = Describe("testing harness agent", func() {
+	var (
+		ctx           context.Context
+		cancel        context.CancelFunc
+		wg            sync.WaitGroup
+		srv           *server.Instance
+		rpcutilAgent  *agentharness.AgentHarness
+		rpcutilClient *rpcutilclient.RpcutilClient
+		brokerLogger  *logrus.Logger
+		brokerLogbuff *gbytes.Buffer
+		serverLogbuff *gbytes.Buffer
+		mockCtl       *gomock.Controller
+		err           error
+	)
+
+	startServerInstance := func(cfgFile string, logbuff *gbytes.Buffer) (*server.Instance, error) {
+		logger := logrus.New()
+		logger.SetOutput(logbuff)
+
+		srv, err = testutil.StartServerInstance(ctx, &wg, cfgFile, logger)
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(logbuff).Should(gbytes.Say("Connected to nats://localhost:4222"))
+
+		return srv, nil
+	}
+
+	createAgent := func(fw inter.Framework) *mcorpc.Agent {
+		ddl, err := fs.FS.ReadFile("ddl/cache/agent/rpcutil.json")
+		Expect(err).ToNot(HaveOccurred())
+
+		rpcutilAgent, err = agentharness.NewWithDDLBytes(fw, mockCtl, "rpcutil", ddl)
+		Expect(err).ToNot(HaveOccurred())
+
+		a, err := rpcutilAgent.Agent()
+		Expect(err).ToNot(HaveOccurred())
+
+		return a
+	}
+
+	createRpcUtilClient := func() (*gbytes.Buffer, *rpcutilclient.RpcutilClient, error) {
+		logBuff, logger := testutil.GbytesLogger(logrus.DebugLevel)
+
+		cfg, err := config.NewConfig("testdata/client.conf")
+		if err != nil {
+			return nil, nil, err
+		}
+
+		cfg.CustomLogger = logger
+		cfg.OverrideCertname = "rip.mcollective"
+
+		fw, err := choria.NewWithConfig(cfg)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		client, err := rpcutilclient.New(fw)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return logBuff, client, nil
+	}
+
+	BeforeEach(func() {
+		mockCtl = gomock.NewController(GinkgoT())
+
+		ctx, cancel = context.WithTimeout(context.Background(), 10*time.Second)
+		DeferCleanup(func() {
+			cancel()
+			Eventually(brokerLogbuff, 5).Should(gbytes.Say("Choria Network Broker shut down"))
+		})
+
+		brokerLogbuff, brokerLogger = testutil.GbytesLogger(logrus.DebugLevel)
+		serverLogbuff = gbytes.NewBuffer()
+
+		_, err := testbroker.StartNetworkBrokerWithConfigFile(ctx, &wg, "testdata/broker.conf", brokerLogger)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(brokerLogbuff, 1).Should(gbytes.Say("Server is ready"))
+
+		srv, err = startServerInstance("testdata/server.conf", serverLogbuff)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = srv.AgentManager().RegisterAgent(ctx, "rpcutil", createAgent(srv.Choria()), srv.Connector())
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(serverLogbuff).Should(gbytes.Say("Registering new agent rpcutil of type rpcutil"))
+
+		_, rpcutilClient, err = createRpcUtilClient()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		mockCtl.Finish()
+	})
+
+	Describe("action stubbing", func() {
```

Comment:
@vjanelle the beef of things here, after in the BeforeEach() adding the test harness into the server pretending to be rpcutil we call it here using the unmodified generated rpcutil client and doing mocks on that `ping` action.